datafusion/common/src/tree_node.rs

```diff
@@ -503,7 +503,7 @@ pub trait TreeNodeVisitor: Sized {
 ///
 /// # See Also:
 /// * [`TreeNode::visit`] to inspect borrowed `TreeNode`s
-pub trait TreeNodeRewriter: Sized {
```

Comment:
Not sure why this was a supertrait of `Sized` but it can't be if we want trait objects